crates/verify/src/etherscan/mod.rs

```diff
@@ -278,8 +278,14 @@ impl EtherscanVerificationProvider {
         builder = if let Some(api_url) = api_url {
             // we don't want any trailing slashes because this can cause cloudflare issues: <https://github.com/foundry-rs/foundry/pull/6079>
             let api_url = api_url.trim_end_matches('/');
-            let base_url = if *verifier_type != VerificationProviderType::Etherscan {
-                // If verifier is not Etherscan then set base url as api url without trialing /api.
+
+            // Verifier is etherscan if explicitly set or if no verifier set (default sourcify) but
+            // API key passed.
+            let is_etherscan = matches!(verifier_type, VerificationProviderType::Etherscan) ||
+                (matches!(verifier_type, VerificationProviderType::Sourcify) &&
```

Comment:
nit: can we replace this with `fn is_etherscan` and `is_sourcify` then this statement becomes easier

Reply:
nice! added in https://github.com/foundry-rs/foundry/pull/10058/commits/22d45608cf12fc2e52fe73f006ad143a020090ec